conda_libmamba_solver/state.py

```diff
@@ -544,21 +544,28 @@ def early_exit(self) -> Dict[str, PackageRecord]:
         """
         sis = self.solver_input_state
         if sis.is_removing:
-            not_installed = [
-                spec for name, spec in sis.requested.items() if name not in sis.installed
-            ]
+            # Make sure that requested packages to be removed match
+            # an installed record. Otherwise, raise an error.
+            # When 'remove --force' is set, remove the package without solving.
+            if sis.force_remove:
+                force_remove_solution = self.current_solution
+            not_installed = []
+            for name, spec in sis.requested.items():
+                for record in sis.installed.values():
+                    if spec.match(record):
+                        if sis.force_remove:
+                            force_remove_solution.remove(record)
+                        break
```

Comment:
Remind me, will this break out of both for loops?

Reply:
Only the innermost one, which is what I intend here. We need to check all requested specs, and see if they are indeed installed. If they are not, then we add it to the list to report the error later.